--- cdf39_1-dist/src/lib/Makefile.orig	2024-09-18 14:42:00.000000000 +0200
+++ cdf39_1-dist/src/lib/Makefile	2025-09-27 12:45:20.349220025 +0200
@@ -113,8 +113,8 @@ libcdf.dylib \
 libcdf.dll \
 libcdf.sl: $(OBJs)
 	@if [ -f $@ ] ; then rm $@ ; else $(NULL) ; fi
-	@echo $(LD) $(LDFLAGS) -o $@ $(OBJs) $(SYSLIBS)
-	      $(LD) $(LDFLAGS) -o $@ $(OBJs) $(SYSLIBS)
+	@echo $(LD) $(LDFLAGS) -o $@ $(OBJs) $(SYSLIBS) -Wl,-soname,libcdf.so.3
+	      $(LD) $(LDFLAGS) -o $@ $(OBJs) $(SYSLIBS) -Wl,-soname,libcdf.so.3
 
 libcdf.o: $(OBJs)
 	@echo $(CC) $(LDFLAGS) -o $@ $(OBJs) $(SYSLIBS)
--- cdf36_4-dist/Makefile.orig	2018-05-16 19:31:40.914229713 +0200
+++ cdf36_4-dist/Makefile	2018-05-16 19:31:55.137562887 +0200
@@ -994,9 +994,9 @@
 
 copy.lib.so:
 	@echo cp $(LIBsrcDIR)/libcdf.so $(INSTALLDIR)/lib
-	      cp $(LIBsrcDIR)/libcdf.so $(INSTALLDIR)/lib/libcdf.$(VERSION).so
+	      cp $(LIBsrcDIR)/libcdf.so $(INSTALLDIR)/lib/libcdf.so.$(VERSION)
 	      rm -f $(INSTALLDIR)/lib/libcdf.so
-	      cd $(INSTALLDIR)/lib && ln -s libcdf.$(VERSION).so libcdf.so
+	      cd $(INSTALLDIR)/lib && ln -s libcdf.so.$(VERSION) libcdf.so.3 && ln -s libcdf.so.$(VERSION) libcdf.so
 
 copy.lib.sl:
 	@echo cp $(LIBsrcDIR)/libcdf.sl $(INSTALLDIR)/lib
